### PR TITLE
Update clustered-indexes.md

### PR DESCRIPTION
**Pull request by tangenta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Update the user documentation of clustered index feature introduced in v5.0-rc & v5.0.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: NA
- Other reference link(s): 
  - clustered index project: https://github.com/pingcap/tidb/projects/45

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: clustered-indexes.md [hunk 1]
```diff
@@ -1,190 +1,215 @@
...
-聚簇索引是 TiDB 在 5.0.0-rc 版本中引入的实验性特性。本文档通过多个示例来说明该特性对 TiDB 查询性能的影响。如需启用此特性及查看详细操作指南，参见 [`tidb_enable_clustered_index` 系统变量介绍](/system-variables.md#tidb_enable_clustered_index-从-v500-rc-版本开始引入)。
+聚簇索引是 TiDB 从 v5.0-rc 开始支持的特性，用于控制含有主键的表数据的存储方式。通过使用聚簇索引，TiDB 可以更好地组织数据表，从而提高某些查询的性能。有些数据库管理系统也将聚簇索引称为“索引组织表” (index-organized tables)。
 
-通过使用聚簇索引，TiDB 可以更好地组织数据表，从而提高某些查询的性能。有些数据库管理系统也将聚簇索引称为“索引组织表” (index-organized tables)。
+目前 TiDB 所有含有主键的表分为两类，分别是：
 
-TiDB 仅支持根据表的`主键`来进行聚簇操作。聚簇索引启用时，`主键`和`聚簇索引`两个术语在一些情况下可互换使用。`主键`指的是约束（一种逻辑属性），而`聚簇索引`描述的是数据存储的物理实现。
+- `NONCLUSTERED`，表示该表的主键为非聚簇索引。在非聚簇索引表中，行数据的键由 TiDB 内部隐式分配的 `_tidb_rowid` 构成，而主键本质上是唯一索引，因此非聚簇索引表存储一行至少需要两个键值对，分别为
+  - `_tidb_rowid` -> row data
+  - primary key columns data -> `_tidb_rowid`
```
Comment (tangenta) [inline clustered-indexes.md:14]:
```suggestion
    - `_tidb_rowid` -> row data
    - primary key columns data -> `_tidb_rowid`
```

File: clustered-indexes.md [hunk 2]
```diff
@@ -1,190 +1,215 @@
...
+CREATE TABLE t (a BIGINT, b VARCHAR(255), PRIMARY KEY(a, b) CLUSTERED);
+CREATE TABLE t (a BIGINT, b VARCHAR(255), PRIMARY KEY(a, b) NONCLUSTERED);
 ```
 
-```sql
-Query OK, 0 rows affected (0.09 sec)
+注意在列定义中的 `KEY` 和 `PRIMARY KEY` 含义相同。
 
-Query OK, 0 rows affected (0.10 sec)
+此外，TiDB 支持[可执行的注释语法](/comment-syntax.md)。：
```
Comment (tangenta) [inline clustered-indexes.md:51]:
```suggestion
此外，TiDB 支持[可执行的注释语法](/comment-syntax.md)：
```

File: clustered-indexes.md [hunk 3]
```diff
@@ -1,190 +1,215 @@
...
-```sql
-Query OK, 0 rows affected (0.00 sec)
+另一类是尚未支持，但未来有计划支持的使用限制：
 
-Query OK, 0 rows affected (0.09 sec)
+- 尚未支持通过 `ALTER TABLE` 语句增加、删除、修改聚簇索引。
+- 在 v5.0-rc 中，如果 `alter-primary-key = true`，无法新建聚簇索引表。从 v5.0 开始，`alter-primary-key` 选项被移除，该限制已被解除。
 
-Query OK, 2 rows affected (0.02 sec)
-Records: 2  Duplicates: 0  Warnings: 0
```
Comment (tangenta) [inline clustered-indexes.md:156]:
```suggestion
```

File: clustered-indexes.md [hunk 4]
```diff
@@ -1,190 +1,214 @@
...
 
-通过使用聚簇索引，TiDB 可以更好地组织数据表，从而提高某些查询的性能。有些数据库管理系统也将聚簇索引称为“索引组织表” (index-organized tables)。
+目前 TiDB 所有含有主键的表分为两类，分别是：
 
-TiDB 仅支持根据表的`主键`来进行聚簇操作。聚簇索引启用时，`主键`和`聚簇索引`两个术语在一些情况下可互换使用。`主键`指的是约束（一种逻辑属性），而`聚簇索引`描述的是数据存储的物理实现。
+- `NONCLUSTERED`，表示该表的主键为非聚簇索引。在非聚簇索引表中，行数据的键由 TiDB 内部隐式分配的 `_tidb_rowid` 构成，而主键本质上是唯一索引，因此非聚簇索引表存储一行至少需要两个键值对，分别为
+    - `_tidb_rowid` -> row data
+    - primary key columns data -> `_tidb_rowid`
+- `CLUSTERED`，表示该表的主键为聚簇索引。在聚簇索引表中，行数据的键由用户给定的主键列数据构成，不存在唯一索引，因此聚簇索引表存储一行至少只要一个键值对，即
+  - primary key columns data -> row data
```
Comment (tangenta) [inline clustered-indexes.md:16]:
```suggestion
    - primary key columns data -> row data
```

File: clustered-indexes.md [hunk 5]
```diff
@@ -1,190 +1,214 @@
 ---
 title: 聚簇索引
-summary: 了解 TiDB 中的聚簇索引。
+summary: 本文档介绍了聚簇索引的概念、使用场景、使用方法、限制和兼容性。
 ---
 
 # 聚簇索引
 
-聚簇索引是 TiDB 在 5.0.0-rc 版本中引入的实验性特性。本文档通过多个示例来说明该特性对 TiDB 查询性能的影响。如需启用此特性及查看详细操作指南，参见 [`tidb_enable_clustered_index` 系统变量介绍](/system-variables.md#tidb_enable_clustered_index-从-v500-rc-版本开始引入)。
+聚簇索引是 TiDB 从 v5.0-rc 开始支持的特性，用于控制含有主键的表数据的存储方式。通过使用聚簇索引，TiDB 可以更好地组织数据表，从而提高某些查询的性能。有些数据库管理系统也将聚簇索引称为“索引组织表” (index-organized tables)。
```
Comment (coocood) [inline clustered-indexes.md:8]: can we remove `rc` and all the obsolete contents?

**Comment by coocood** [comment]
LGTM

File: clustered-indexes.md [hunk 6]
```diff
@@ -1,190 +1,211 @@
...
 
-启用聚簇索引后，主键替代 64 位的 `handle` 值成为表中每行数据的内部指针，所以对存储空间的需求可能会上升，尤其当表中包含很多二级索引时。以下表为例：
+聚簇索引仅与 v5.0 及以后版本的以下生态工具兼容：
+
+- 备份与恢复工具 BR、Dumpling、Lightning。
+- 数据迁移和同步工具 DM、TiCDC。
+
+v5.0 的 BR 不能通过备份恢复将非聚簇索引表转换成聚簇索引表，反之亦然。
+
+### 与 TiDB 其他特性的兼容性
```
Comment (TomShawn) [inline clustered-indexes.md:182]: @tangenta @yikeke @qiancai 这个文档涉及 v5.0 版本前后的兼容性问题，看最新开发版文档的用户应该是不需要这个信息的吧？

File: clustered-indexes.md [hunk 7]
```diff
@@ -1,190 +1,211 @@
...
-1 row in set (0.00 sec)
-```
+TiDB 支持对聚簇索引表的升级兼容，但不支持降级兼容，即高版本 TiDB 聚簇索引表的数据在低版本 TiDB 上不可用。
+
+聚簇索引在 TiDB v3.0 和 v4.0 中已完成部分支持，当表中存在单个整数列作为主键时默认启用，即：
+
+- 表设置了主键
+- 主键只有一列
+- 主键的数据类型为整数类型
+
```
[the remark this answers is not in the document]
Comment (tangenta) [reply clustered-indexes.md:166]: 但是我理解“升降级兼容性”就是讨论版本的？

File: clustered-indexes.md [hunk 8]
```diff
@@ -1,190 +1,212 @@
...
+## 限制
 
-Query OK, 2 rows affected (0.02 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+目前 TiDB 的聚簇索引具有以下两类限制：
 
-Query OK, 2 rows affected (0.01 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+- 明确不支持且没有支持计划的使用限制：
+    - 不支持与 TiDB Binlog 一起使用。开启 Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)。
```
Comment (TomShawn) [inline clustered-indexes.md:136]:
```suggestion
    - 不支持与 TiDB Binlog 一起使用。开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)。
```

File: clustered-indexes.md [hunk 9]
```diff
@@ -1,190 +1,212 @@
...
 
-Query OK, 2 rows affected (0.02 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+目前 TiDB 的聚簇索引具有以下两类限制：
 
-Query OK, 2 rows affected (0.01 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+- 明确不支持且没有支持计划的使用限制：
+    - 不支持与 TiDB Binlog 一起使用。开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)。
+    - 不支持与 `SHARD_ROW_ID_BITS` 一起使用。
```
Comment (qiancai) [inline clustered-indexes.md:137]: https://docs.google.com/document/d/1AgqZtzXr1Oqdbq_eNdPohE2tEoqE2mgQ11SsadSXQWw/edit# 里提到了 “聚簇索引不支持与 `SHARD_ROW_ID_BITS`,`AUTO_RANDOM`，`PRE_SPLIT_REGIONS` 属性一起使用。”， 所以这里是否需要补充 `AUTO_RANDOM`，`PRE_SPLIT_REGIONS` 属性。

Comment (tangenta) [reply clustered-indexes.md]: PRE_SPLIT_REGIONS 描述已更新。关于 AUTO_RANDOM 属性的描述已更新在文档最下方。

File: clustered-indexes.md [hunk 10]
```diff
@@ -1,190 +1,219 @@
...
 
-Query OK, 2 rows affected (0.02 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+目前 TiDB 的聚簇索引具有以下两类限制：
 
-Query OK, 2 rows affected (0.01 sec)
-Records: 2  Duplicates: 0  Warnings: 0
+- 明确不支持且没有支持计划的使用限制：
+    - 不支持与 TiDB Binlog 一起使用。开启 TiDB Binlog 后 TiDB 不允许创建非单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)。
+    - 不支持与 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) 一起使用；[`PRE_SPLIT_REGIONS`](sql-statements/sql-statement-split-region.md#pre_split_regions) 在聚簇索引表上不生效。
```
Comment (TomShawn) [inline clustered-indexes.md:137]:
```suggestion
    - 不支持与 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) 一起使用；[`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions) 在聚簇索引表上不生效。
```

File: clustered-indexes.md [hunk 11]
```diff
@@ -1,190 +1,211 @@
...
+### MySQL 兼容性
 
-在 MySQL 中，InnoDB 存储引擎默认会使用任意主键作为聚簇索引，此处行为与之一致。
+TiDB 支持使用可执行注释的语法来包裹 `CLUSTERED` 或 `NONCLUSTERED` 关键字，且 `SHOW CREATE TABLE` 的结果均包含 TiDB 特有的可执行注释，因此这部分 DDL 语句能被 MySQL 或低版本的 TiDB 识别并执行。
 
-## 存储需求
+### TiDB 生态工具兼容性
 
-启用聚簇索引后，主键替代 64 位的 `handle` 值成为表中每行数据的内部指针，所以对存储空间的需求可能会上升，尤其当表中包含很多二级索引时。以下表为例：
+聚簇索引仅与 v5.0 及以后版本的以下生态工具兼容：
```
[the remark this answers is not in the document]
Comment (qiancai) [reply clustered-indexes.md:176]: /lgtm

File: clustered-indexes.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (yikeke) [reply clustered-indexes.md:167]: @TomShawn 这里的目的就是讨论低版本和高版本的兼容性，对于从低版本升上来的用户，这个版本信息很重要，保留即可

File: clustered-indexes.md [hunk 12]
```diff
@@ -1,190 +1,211 @@
...
-```
+TiDB 支持对聚簇索引表的升级兼容，但不支持降级兼容，即高版本 TiDB 聚簇索引表的数据在低版本 TiDB 上不可用。
+
+聚簇索引在 TiDB v3.0 和 v4.0 中已完成部分支持，当表中存在单个整数列作为主键时默认启用，即：
+
+- 表设置了主键
+- 主键只有一列
+- 主键的数据类型为整数类型
+
+从 TiDB v5.0 开始，不论是单整数列主键还是其他类型主键，默认均为非聚簇索引。该行为变更可能导致默认配置下的 TiDB 在某些场景中出现性能回退，此时可以考虑显式启用聚簇索引。
```
[the remark this answers is not in the document]
Comment (yikeke) [reply clustered-indexes.md:168]: 保留即可

File: clustered-indexes.md (hunk 6 above)
Comment (qiancai) [reply clustered-indexes.md:183]: 因为这个改动涉及到 v5.0 用户行为的改变，所以我觉得需要保留

File: clustered-indexes.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (TomShawn) [reply clustered-indexes.md:167]: OK

File: clustered-indexes.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (yikeke) [reply clustered-indexes.md:176]: 赞同保留

**Pull request merged**